### PR TITLE
Uniquely index event-type with teamId and slug

### PR DESCRIPTION
**Pull request by afzalsayed96**
## What does this PR do?

We are able to create multiple event-types with the same `slug` for the same team. This should be disallowed as all even-types pages are uniquely identified with their team name & `slug`. This causes a collision and results in a 404 page.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## How should this be tested?

Try creating multiple events with the same slug in the same team.

### CODE REVIEW

**Comment by afzalsayed96** [comment]
Note: Same issue exists for user level team creation. However, unlike team, users and event-types are linked with implicit [many-to-may relationships](https://www.prisma.io/docs/concepts/components/prisma-schema/relations/many-to-many-relations#implicit-many-to-many-relations). So a `[userId, slug]` index does not work in this case.

What would be the ideal way to solve this?

**Comment by afzalsayed96** [comment]
Can someone help me understand the reasoning behind making `users` as many-to-many relationship instead of many-to-one `user` relationship similar to `team`?

File: packages/prisma/migrations/20220409195425_index_event_types_team_id_slug/migration.sql [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+
+*/
+-- CreateIndex
+CREATE UNIQUE INDEX "EventType_teamId_slug_key" ON "EventType"("teamId", "slug");
+
+-- RenameIndex
+ALTER INDEX "VerificationRequest.identifier_token_unique" RENAME TO "VerificationRequest_identifier_token_key";
+
+-- RenameIndex
+ALTER INDEX "VerificationRequest.token_unique" RENAME TO "VerificationRequest_token_key";
```
Comment (zomars) [inline packages/prisma/migrations/20220409195425_index_event_types_team_id_slug/migration.sql:14]: These renames should be handled in another PR since these have noting to do with this change:
```suggestion
```

**Review by zomars — APPROVED** [review]
